src/hooks/useNavbarToggle.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef, useState } from "react"
+
+const useNavbarToggle = () => {
+  const [isToggle, setIsToggle] = useState(false)
+  const toggleRef = useRef<HTMLButtonElement>(null)
+
+  const handleOutSideClick = (e: Event) => {
+    const current = toggleRef.current
+    if (isToggle && current && !current.contains(e.target as Node)) {
+      setIsToggle(false)
+    }
+  }
+
+  const handleToggle = () => {
+    setIsToggle((prevIsToggle: boolean) => !prevIsToggle)
```

Comment:
[P2] 요 부분의 타입 정의도 필요없을것 같네요!
```suggestion
    setIsToggle((prevIsToggle) => !prevIsToggle)
```